contracts/core/EntryPoint.sol

```diff
@@ -141,6 +149,24 @@ contract EntryPoint is IEntryPoint, StakeManager {
         _compensate(beneficiary, collected);
     }
 
+    function simulateExecution(UserOperation calldata op) external override {
+
+        UserOpInfo memory opInfo;
+
+        (uint256 deadline, uint256 paymasterDeadline,) = _validatePrepayment(0, op, opInfo, SIMULATE_FIND_AGGREGATOR);
+        //ignore signature check failure
+        if (deadline == SIG_VALIDATION_FAILED) {
+            deadline = 0;
+        }
+        if (paymasterDeadline == SIG_VALIDATION_FAILED) {
+            paymasterDeadline = 0;
+        }
+        _validateDeadline(0, opInfo, deadline, paymasterDeadline);
+        numberMarker();
+        uint256 paid = _executeUserOp(0, op, opInfo);
+        revert ExecutionComplete(opInfo.preOpGas, paid, deadline, paymasterDeadline);
```

Comment:
You don't know if the original `deadline` was `0` or `SIG_VALIDATION_FAILED` here. If that is not important there is no point even emitting those values, is there?

Reply:
I don't know what is the value, but I do want the bundler (external code) to know it.